### PR TITLE
Add a configurable prefix when an object is imported or created (VM, SR)

### PR DESCRIPTION
**Pull request by gduperrey**
Signed-off-by: Gael Duperrey <gduperrey@vates.fr>

### CODE REVIEW

File: lib/common.py [hunk 1]
```diff
@@ -16,6 +17,17 @@ def vm_image(vm_key):
...
     return url
 
+def objects_name_prefix():
+    try:
+        from data import OBJECTS_NAME_PREFIX
+        name_prefix = OBJECTS_NAME_PREFIX
+    except ImportError:
+        name_prefix = f"[{os.getlogin()}]"
+        if not name_prefix:
+            name_prefix = '[TEST]'
```
Comment (stormi) [inline lib/common.py:27]: Is it possible that name_prefix is not defined at this stage?

File: lib/common.py [hunk 2]
```diff
@@ -16,6 +17,17 @@ def vm_image(vm_key):
         url = DEF_VM_URL + url
     return url
 
+def objects_name_prefix():
+    try:
+        from data import OBJECTS_NAME_PREFIX
+        name_prefix = OBJECTS_NAME_PREFIX
+    except ImportError:
+        name_prefix = f"[{os.getlogin()}]"
```
Comment (stormi) [inline lib/common.py:25]: According to https://docs.python.org/3/library/os.html#os.getlogin `getpass.getuser()` should be preferred over `os.get_login()`.

File: data.py-dist [hunk 3]
```diff
@@ -6,6 +6,11 @@
 HOST_DEFAULT_USER = "root"
 HOST_DEFAULT_PASSWORD = ""
 
+# The following prefix will be added to the `name-label` parameter of XAPI objects
+# that the tests will create or import, such as  VMs and SRs.
+# Default value: [TEST]
```
Comment (stormi) [inline data.py-dist:11]: Not true anymore.

**Review by stormi — CHANGES_REQUESTED** [review]
In the commit message, s/launch/started/

File: lib/common.py (hunk 1 above)
Comment (gduperrey) [reply lib/common.py:27]: i don't know, but in case we didn't catch a user name, it will be cover.

Comment (stormi) [reply lib/common.py:27]: The docs for os.getlogin or getpass.getuser don't mention a possible empty value, and I believe that if they really can't get the user name, they'll raise an Exception anyway.

**Review by benjamreis — APPROVED** [review]

File: data.py-dist [hunk 4]
```diff
@@ -6,6 +6,11 @@
 HOST_DEFAULT_USER = "root"
 HOST_DEFAULT_PASSWORD = ""
 
+# The following prefix will be added to the `name-label` parameter of XAPI objects
+# that the tests will create or import, such as  VMs and SRs.
```
Comment (stormi) [inline data.py-dist:10]:
```suggestion
# that the tests will create or import, such as VMs and SRs.
```

File: lib/common.py [hunk 5]
```diff
@@ -16,6 +17,15 @@ def vm_image(vm_key):
         url = DEF_VM_URL + url
     return url
 
+def objects_name_prefix():
+    try:
+        from data import OBJECTS_NAME_PREFIX
+        name_prefix = OBJECTS_NAME_PREFIX
+    except ImportError:
+        name_prefix = f"[{getpass.getuser()}]"
+    finally:
```
Comment (stormi) [inline lib/common.py:26]: I should have mentioned it earlier, as I had seen it yesterday already: this `finally` statement not only is not necessary, but it's dangerous. It's unlikely, but if an exception was raised that is not ImportError and would prevent name_prefix from being set, we would execute `return name_prefix` on a variable that is not set. Without `finally`, the `return` statement will only be executed if name_prefix is set.

**Review by stormi — CHANGES_REQUESTED** [review]

**Review by benjamreis — APPROVED** [review]

**Review by stormi — APPROVED** [review]

File: lib/common.py [hunk 6]
```diff
@@ -16,6 +17,15 @@ def vm_image(vm_key):
         url = DEF_VM_URL + url
     return url
 
+def prefix_object_name(label):
+    try:
+        from data import OBJECTS_NAME_PREFIX
+        name_prefix = OBJECTS_NAME_PREFIX
+    except ImportError:
+        name_prefix = f"[{getpass.getuser()}]"
+    object_name = f"{name_prefix} {label}"
```
Comment (stormi) [inline lib/common.py:26]: you don't need the `object_name` variable here as you don't use it for anything but returning it. You can return `f"{name_prefix} {label}"` directly.

**Pull request merged**